src/partition/ReceivedPartition.cpp

```diff
@@ -610,10 +614,19 @@ void ReceivedPartition::createOwnerInformation()
   }
 }
 
-bool ReceivedPartition::areProvidedMeshesEmpty() const
+bool ReceivedPartition::isAnyProvidedMeshNonEmpty() const
 {
-  return (_fromMapping && not _fromMapping->getOutputMesh()->vertices().empty()) ||
-         (_toMapping && not _toMapping->getInputMesh()->vertices().empty());
+  for (mapping::PtrMapping fromMapping : _fromMappings) {
```

Comment:
Same as above, but also don't need write access.
Thus we can use a const reference.
```suggestion
  for (const auto& fromMapping : _fromMappings) {
```